src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,333 @@
+                }
+                this.max_chunk_limit = max_chunk_limit;
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean isListString(Object value) {
+        // an empty list is also List<String>
+        if (!(value instanceof List)) {
+            return false;
+        }
+        for (Object element : (List<Object>) value) {
+            if (!(element instanceof String)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<String> chunkString(String content) {
+        FieldChunker chunker = ChunkerFactory.create(chunkerType, analysisRegistry);
+        List<String> result = chunker.chunk(content, chunkerParameters);
+        current_chunk_count += result.size();
+        if (max_chunk_limit != DEFAULT_MAX_CHUNK_LIMIT && current_chunk_count > max_chunk_limit) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as the number of chunks ["
+                    + current_chunk_count
+                    + "] exceeds the maximum chunk limit ["
+                    + max_chunk_limit
+                    + "]"
+            );
+        }
+        return result;
+    }
+
+    private List<String> chunkList(List<String> contentList) {
+        // flatten the List<List<String>> output to List<String>
+        List<String> result = new ArrayList<>();
+        for (String content : contentList) {
+            result.addAll(chunkString(content));
+        }
+        return result;
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<String> chunkLeafType(Object value) {
+        // leaf type is either String or List<String>
+        List<String> chunkedResult = null;
+        if (value instanceof String) {
+            chunkedResult = chunkString(String.valueOf(value));
+        } else if (isListString(value)) {
+            chunkedResult = chunkList((List<String>) value);
+        }
+        return chunkedResult;
+    }
+
+    /**
+     * This method will be invoked by PipelineService to perform chunking and then write back chunking results to the document.
+     * @param ingestDocument {@link IngestDocument} which is the document passed to processor.
+     */
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        validateFieldsValue(ingestDocument);
+        current_chunk_count = 0;
+        if (Objects.equals(chunkerType, FIXED_LENGTH_ALGORITHM)) {
+            // add maxTokenCount setting from index metadata to chunker parameters
+            Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+            String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
+            int maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(environment.settings());
+            IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
```

Comment:
Will these two maxTokenCount have different value?

Reply:
It is possible to have different value. If the index is not defined,  we take the default value of maxTokenCount from the cluster environment. If user specify an available index name, we will look up `index.analyze.max_token_count` from the index settings.